elodie/filesystem.py

```diff
@@ -320,7 +320,35 @@ def parse_mask_for_location(self, mask, location_parts, place_name):
 
         return folder_name
 
+    def process_checksum(self, _file, allow_duplicate):
+        db = Db()
+        checksum = db.checksum(_file)
+        if(checksum is None):
+            log.info('Could not get checksum for %s. Skipping...' % _file)
+            return None
+
+        # If duplicates are not allowed then we check if we've seen this file
+        #  before via checksum. We also check that the file exists at the
+        #   location we believe it to be.
+        # If we find a checksum match but the file doesn't exist where we
+        #  believe it to be then we write a debug log and proceed to import.
+        checksum_file = db.get_hash(checksum)
+        if(allow_duplicate is False and checksum_file is not None):
+            if(os.path.isfile(checksum_file)):
+                log.info('%s already exists at %s. Skipping...' % (
+                    _file,
+                    checksum_file
+                ))
+                return None
+            else:
+                log.info('%s matched checksum but file not found at %s. Importing again...' % (  # noqa
```

Comment:
I would change the log message here to be `'%s matched checksum but file not found at %s.`